coalib/bears/Bear.py

```diff
@@ -22,6 +24,34 @@
 from .meta import bearclass
 
 
+class db(pdb.Pdb):
+    def __init__(self, *args, **kwargs):
+        self.do_q = self.custom_quit
+        self.do_quit = self.custom_quit
+        self.do_exit = self.custom_quit
```

Comment:
I think you change this per-instance-patching to a per-class-patching (then you can completely leave out `__init__`, as you've nothing more to do there):
```python
class db(pdb.Pdb):
    def do_quit(self, arg):
        self.clear_all_bears()
        super().do_continue(arg)  # <-- super is actually redundant, but you can keep it since it makes the intention clear that we want to use the old, known "continue"
        return 1

    do_q = do_quit
    do_exit = do_quit
```